gatsby-node.js

```diff
@@ -289,18 +289,57 @@ async function getIndexToUse({ index, tempIndex, enablePartialUpdates }) {
   return index;
 }
 
-async function getSettingsToApply({ settings, index, tempIndex, indexToUse }) {
-  const requestedSettings = settings ? settings : await index.getSettings();
+
+async function getSettingsToApply({
+  settings: givenSettings,
+  index,
+  tempIndex,
+  indexToUse,
+}) {
+  const { replicaUpdateMode, ...settings } = givenSettings;
+  var existingSettings = {};
+  try {
+    existingSettings = await index.getSettings();
+  } catch (e) {
+    report.error(`${e.toString()} ${index.indexName}`);
+  }
+  const replicasToSet = getReplicasToSet(
+    settings.replicas,
+    existingSettings.replicas,
+    replicaUpdateMode
+  );
+  var requestedSettings = settings ? settings : existingSettings;
 
   // If we're building replicas, we don't want to add them to temporary indices
   if (indexToUse === tempIndex) {
     const { replicas, ...adjustedSettings } = requestedSettings;
-    return adjustedSettings;
+    requestedSettings = adjustedSettings;
+    if (replicasToSet)
+      await index.setSettings({ replicas: replicasToSet }).catch((err) => {
+        report.panic(`${err.toString()} ${replicasToSet}`);
+      });
   }
-
   return requestedSettings;
 }
 
+function getReplicasToSet(givenReplicas, existingReplicas, replicaUpdateMode) {
+  const obj = {};
+  if (replicaUpdateMode == 'replace')
+    return (
+      (givenReplicas &&
+        givenReplicas.filter((r) => r && typeof r == 'string')) ||
+      []
+    );
```

Comment:
```suggestion
  if (replicaUpdateMode == 'replace')
    return givenReplicas;
```

why is that not just this? Won't the replica value always be an array of strings?